Composer/packages/client/src/recoilModel/dispatchers/navigation.ts

```diff
@@ -50,44 +50,37 @@ export const navigationDispatcher = () => {
 
   const navTo = useRecoilCallback(
     ({ snapshot, set }: CallbackInterface) => async (
-      projectId: string,
       skillId: string | null,
-      dialogId: string,
+      dialogId: string | null,
       breadcrumb: BreadcrumbItem[] = []
     ) => {
-      const dialogs = await snapshot.getPromise(dialogsSelectorFamily(projectId));
-      const designPageLocation = await snapshot.getPromise(designPageLocationState(projectId));
-      const updatedBreadcrumb = cloneDeep(breadcrumb);
-      set(currentProjectIdState, projectId);
+      const rootBotProjectId = await snapshot.getPromise(rootBotProjectIdSelector);
+      if (rootBotProjectId == null) return;
 
-      let path;
-      if (dialogId !== designPageLocation.dialogId) {
-        const currentDialog = dialogs.find(({ id }) => id === dialogId);
-        const beginDialogIndex = currentDialog?.triggers.findIndex(({ type }) => type === SDKKinds.OnBeginDialog);
+      const projectId = skillId ?? rootBotProjectId;
 
-        if (typeof beginDialogIndex !== 'undefined' && beginDialogIndex >= 0) {
-          path = createSelectedPath(beginDialogIndex);
-          path = encodeArrayPathToDesignerPath(currentDialog?.content, path);
-          updatedBreadcrumb.push({ dialogId, selected: '', focused: '' });
-        }
-      }
-
-      const currentUri = convertPathToUrl(projectId, skillId, dialogId, path);
+      const designPageLocation = await snapshot.getPromise(designPageLocationState(projectId));
+      set(currentProjectIdState, projectId);
 
-      if (checkUrl(currentUri, projectId, designPageLocation)) return;
+      const currentUri = convertPathToUrl(rootBotProjectId, projectId, dialogId);
+      if (checkUrl(currentUri, rootBotProjectId, projectId, designPageLocation)) return;
 
-      navigateTo(currentUri, { state: { breadcrumb: updatedBreadcrumb } });
+      navigateTo(currentUri, { state: { breadcrumb } });
     }
   );
 
   const selectTo = useRecoilCallback(
     ({ snapshot, set }: CallbackInterface) => async (
-      projectId: string,
       skillId: string | null,
       destinationDialogId: string | null,
       selectPath: string
     ) => {
       if (!selectPath) return;
+      const rootBotProjectId = await snapshot.getPromise(rootBotProjectIdSelector);
+      if (rootBotProjectId == null) return;
+
+      const projectId = skillId ?? rootBotProjectId;
+
       set(currentProjectIdState, projectId);
```

Comment:
CurrentProjectIdstate is set wrongly here. It should be the skillId if there its a skill or the rootbotId

Reply:
That's what line 82 there does. If skillId is not null, it uses that, otherwise it defaults to the root ID.